:book: Adds section on envtest limitations (#2857)
kubernetes-sigs/kubebuilder: docs/book/src/reference/envtest.md

Expect(deployment.ObjectMeta.OwnerReferences).To(ContainElement(expectedOwnerReference))
```

> Another limitation worth knowing about is that envtest does not support deletion of namespaces. Deleting a namespace will seem to succeed, but the namespace will just be put in Terminating state, and never actually be reclaimed. Trying to recreate the namespace will fail. For this reason, it is not advisable to try to create and delete namespaces as a mechanism to isolate tests from each other. To overcome this limitation, create a new namespace for each test.
> 
> Even so, when one test completes (in "namespace-1") and another test starts (in "namespace-2"), the controller will still be reconciling any active objects from "namespace-1". This can be avoided by ensuring that all tests clean up after themselves as part of the test teardown.  If teardown of a namespace is difficult, it may be possible to wire the controller in such a way that it ignores reconcile requests that come from namespaces other than the one being tested.

Reviewer: ```suggestion

<aside class="warning">

<h2>Namesapce usage limitation</h2>

Env Test does not support namespace deletion. Deleting a namespace will seem to succeed, but the namespace will just be put in a Terminating state, and never actually be reclaimed. Trying to recreate the namespace will fail. 

To overcome this limitation you can: 

Create a new namespace for each test. Even so, when one test completes (into i.e. "namespace-1") and another test starts (into i.e. "namespace-2"), the controller will still be reconciling any active objects from "namespace-1". This can be avoided by ensuring that all tests clean up after themselves as part of the test teardown.  If teardown of a namespace is difficult, it may be possible to wire the controller in such a way that it ignores reconcile requests that come from namespaces other than the one being tested.

For further information see the issue raised in the controller-runtime [controller-runtime/issues/880](https://github.com/kubernetes-sigs/controller-runtime/issues/880) to add this support.

</aside>
```

Reviewer: @mogsie,

Thank you for the contribution 🥇. Just a few nits, see the above suggestion:

a) Could you please add in the description, closes: https://github.com/kubernetes-sigs/kubebuilder/issues/2856 (we need to look for it in C+R)
b) See that is good we add the issue in C+R to add this support
c) Also, it seems fit better in a note also we need to put after the code

> 
> ```go
> 
> ```

Reviewer: ```suggestion
```

Remove ^

> 

Reviewer: ```suggestion

<aside class="warning">

<h2>Namesapce usage limitation</h2>

Env Test does not support namespace deletion. Deleting a namespace will seem to succeed, but the namespace will just be put in a Terminating state, and never actually be reclaimed. Trying to recreate the namespace will fail. 

To overcome this limitation you can: 

Create a new namespace for each test. Even so, when one test completes (into i.e. "namespace-1") and another test starts (into i.e. "namespace-2"), the controller will still be reconciling any active objects from "namespace-1". This can be avoided by ensuring that all tests clean up after themselves as part of the test teardown.  If teardown of a namespace is difficult, it may be possible to wire the controller in such a way that it ignores reconcile requests that come from namespaces other than the one being tested.

</aside>
```

Reviewer: @mogsie, 

Thank you for the contribution 🥇. Just a few nits, see the above suggestion:

a) Could you please add in the description, closes: https://github.com/kubernetes-sigs/kubebuilder/issues/2856 (we need to look for it in C+R)
b) See that is good we add the issue in C+R to add this support
c) Also, it seems fit better in a note also we need to put after the code
